instrumentation/mongodb-reactive-streams-4.2/src/main/java/com/nr/agent/mongo/NRCallbackWrapper.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *  * Copyright 2023 New Relic Corporation. All rights reserved.
+ *  * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package com.nr.agent.mongo;
+
+import com.mongodb.internal.async.SingleResultCallback;
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.api.agent.DatastoreParameters;
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.Segment;
+import com.newrelic.api.agent.Token;
+import com.newrelic.api.agent.Trace;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class NRCallbackWrapper<T> implements SingleResultCallback<T> {
+
+    private static final AtomicBoolean isNotTransformed = new AtomicBoolean(true);
+    public Token token = null;
+    public Segment segment = null;
+    public DatastoreParameters params = null;
+    private SingleResultCallback<T> delegate = null;
+
+    public NRCallbackWrapper(SingleResultCallback<T> d) {
+        delegate = d;
+        if (isNotTransformed.get()) {
+            AgentBridge.instrumentation.retransformUninstrumentedClass(getClass());
+            isNotTransformed.set(false);
+        }
```

Comment:
You might not need this logic if you change the package this class is in.

Reply:
For some reason, changing the package did not work as intended. It did instrument the class, but when the time came to link the token it said that no transaction was available, even though the running bytecode had the @Trace(async=true) annotation and the exit tracer was being created.